client/src/pages/Dashboard/EventModal/EventModal.tsx

```diff
@@ -1,57 +1,87 @@
+import { useAuth } from '../../../context/useAuthContext';
 
 import useStyles from './useStyles';
 
 interface Props {
+  fetchMeetingsCallback: (id: string) => void;
   open: boolean;
   setOpen: React.Dispatch<React.SetStateAction<boolean>>;
 }
 
-interface IForm {
-  userId: string;
-  duration: string;
-}
-
-export default function EventModal({ open, setOpen }: Props): JSX.Element {
+export default function EventModal({ fetchMeetingsCallback, open, setOpen }: Props): JSX.Element {
   const classes = useStyles();
   const { updateSnackBarMessage } = useSnackBar();
+  const loggedInUser: any = useAuth().loggedInUser;
 
-  const [form, setForm] = useState<IForm>({ userId: '', duration: '' });
+  const [name, setName] = useState<string>('');
+  const [duration, setDuration] = useState<number | string | unknown>('');
 
   const handleClose = () => setOpen(false);
 
-  const handleChange = (e: React.ChangeEvent<{ value: unknown }>) => {
-    setForm({ ...form, duration: e.target.value as string });
+  const handleChangeName = (e: { target: HTMLInputElement | HTMLTextAreaElement }) => {
+    setName(e.target.value);
   };
 
-  console.log(form);
+  const handleChangeSelect = (e: React.ChangeEvent<{ value: unknown }>) => {
+    setDuration(e.target.value);
+  };
 
   const handleClick = async () => {
-    const { duration } = form;
-
-    if (duration === '') {
+    if (duration === '' || name === '') {
       updateSnackBarMessage('Please enter a valid event duration');
     }
 
     const res = await fetch('/meeting', {
       method: 'POST',
       credentials: 'include',
-      body: JSON.stringify(form),
+      body: JSON.stringify({
+        userId: loggedInUser._id,
+        name: name,
+        duration: duration,
+      }),
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      },
     });
+
+    if (res.status === 400) {
+      const message = await res.json();
+      updateSnackBarMessage(message.error);
+      return;
+    }
+
+    if (res.status === 406) {
+      const message = await res.json();
+      updateSnackBarMessage(message.error);
+      return;
+    }
+
+    fetchMeetingsCallback(loggedInUser._id);
+    setName('');
+    setDuration('false');
+    setOpen(false);
   };
 
   const modalBody = (
     <div className={classes.paper}>
       <Box className={classes.formHeader}>
         <h1>Create New Event</h1>
       </Box>
+      <Box className={classes.formItem}>
+        <h3>Enter the name of the event:</h3>
+        <FormControl>
+          <OutlinedInput name="name" placeholder="Name" onChange={(e) => handleChangeName(e)} />
+        </FormControl>
+      </Box>
       <Box className={classes.formItem}>
         <h3>Enter the duration of the event:</h3>
         <FormControl>
           <InputLabel>Duration</InputLabel>
-          <Select value={form.duration} onChange={handleChange} className={classes.formInput}>
```

Comment:
We should probably use Formik when we are making forms, it will be a bit more maintainable and it's available in the project.

Reply:
Hey Ethan, I just implemented Formik. I wish I had looked into it sooner, it makes dealing with forms so much easier!